### PR TITLE
WIP: Add sub-packages for kubeadm and kubectl.

### PR DESCRIPTION
**Pull request by dgoodwin**
Installing kubeadm or the CNI plugin will alter the kubelet options used
to run accordingly.

### CODE REVIEW

**Comment by dgoodwin** [comment]
Just publishing this WIP to demonstrate the logic in kubelet-wrapper and hopefully get some feedback. Basically just checking if there's an env file for kubeadm/plugin-cni, and if so sourcing them and appending to the KUBELET_OPTS when we run.

The StartLimitInterval change is not great, will discuss on call today but it spams the logs like crazy. I can't find any way to slow it down that doesn't cause it to stop restarting eventually.

**Comment by dgoodwin** [comment]
This is an attempt to sync up with recent work done for the debian packages in #85 and #81.

**Comment by dgoodwin** [comment]
@mikedanese @errordeveloper let me know what you guys think if time allows. Still doing some testing and it's currently using local binaries instead of curling just while I'm working on it. (and pending a published url for kubeadm)

**Comment by dgoodwin** [comment]
I'm publishing my WIP rpms to a repository, you can use by adding this to kubelet.repo file in /etc/yum.repos.d:

```
[kubelet]
name=kubelet
baseurl=http://files.rm-rf.ca/rpms/kubelet/
enabled=1
gpgcheck=0
```

**Comment by dgoodwin** [comment]
Updated with a bunch of changes to sync up with Debian packaging and latest discussions.

**Comment by dgoodwin** [comment]
This has been updated to no longer use Docker dependencies, and with a hand built kubeadm binary with the selinux fix. The only things remaining are to start curling kubeadm once published, and adjust the version of kubectl and kubelet from 1.4.0-beta.8 to final form.

**Comment by dgoodwin** [comment]
RPMs in the repo above are updated.

**Comment by dgoodwin** [comment]
Another update, now pulling kubeadm from published location, bumped to beta.11 packages.

File: rpm/10-kubeadm.conf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[Service]
+Environment="KUBELET_KUBECONFIG_ARGS=--kubeconfig=/etc/kubernetes/kubelet.conf --require-kubeconfig=true"
+Environment="KUBELET_SYSTEM_PODS_ARGS=--pod-manifest-path=/etc/kubernetes/manifests --allow-privileged=true"
+Environment="KUBELET_NETWORK_ARGS=--network-plugin=cni --cni-conf-dir=/etc/cni/net.d --cni-bin-dir=/opt/cni/bin"
+Environment="KUBELET_DNS_ARGS=--cluster-dns=100.64.0.10 --cluster-domain=cluster.local"
+Environment="KUBELET_EXTRA_ARGS=--v=4"
+ExecStart=
+ExecStart=/usr/bin/kubelet $KUBELET_KUBECONFIG_ARGS $KUBELET_SYSTEM_PODS_ARGS $KUBELET_NETWORK_ARGS $KUBELET_DNS_ARGS $KUBELET_EXTRA_ARGS
```
Comment (luxas) [inline rpm/10-kubeadm.conf]: nit: on debian kubelet is in `/usr/sbin`

Comment (luxas) [inline rpm/10-kubeadm.conf]: But we can fix this later

File: rpm/kubelet.spec [hunk 2]
```diff
@@ -21,14 +21,34 @@ Requires: ethtool
...
+
+Summary: Command-line utility for interacting with a Kubernetes cluster.
+
+%description -n kubectl
+Command-line utility for interacting with a Kubernetes cluster.
+
+%package -n kubeadm
+
+Version: 1.5.0
+Release: %{RPM_RELEASE}.alpha.gc19e08e
```
Comment (luxas) [inline rpm/kubelet.spec]: should be `%{RPM_RELEASE}.alpha.0-1403-gc19e08e`

If `-` are unsupported, then `.alpha.0.1403.gc19e08e`

Comment (dgoodwin) [inline rpm/kubelet.spec]: Updated.

File: rpm/kubelet.spec [hunk 3]
```diff
@@ -21,14 +21,34 @@ Requires: ethtool
...
+%package -n kubernetes-cni
 
+Version: 0.3.0.1
+Release: %{RPM_RELEASE}.07a8a2
 Summary: Binaries required to provision kubernetes container networking
 Requires: kubelet
 
-%description plugin-cni
+%description -n kubernetes-cni
 Binaries required to provision container networking.
```
Comment (errordeveloper) [inline rpm/kubelet.spec]: Basic CNI network plugins for Kubernetes.

File: rpm/kubelet.spec [hunk 4]
```diff
@@ -21,14 +21,34 @@ Requires: ethtool
...
+Summary: Command-line utility for interacting with a Kubernetes cluster.
+
+%description -n kubectl
+Command-line utility for interacting with a Kubernetes cluster.
+
+%package -n kubeadm
+
+Version: 1.5.0
+Release: %{RPM_RELEASE}.alpha.0.1403.gc19e08e
+Summary: Command-line utility for administering a Kubernetes cluster. (ALPHA)
```
Comment (errordeveloper) [inline rpm/kubelet.spec]: May be good to make these consistent with Debian...

**Comment by luxas** [comment]
@mikedanese Mind merging this so we can keep moving forward?

**Pull request merged**